File: myBackend/app/models/categoryModel.py
from pydantic import BaseModel
from enum import Enum


class CategoryType(str, Enum):
    INCOME = 'income'
    EXPENSE = 'expense'


class CategoryIn(BaseModel):
    name: str
    category_type: CategoryType


class CategoryOut(BaseModel):
    category_id: int
    name: str
    category_type: CategoryType

File: my_frontend/lib/models/category_model.dart
class CategoryModel {
  final int categoryId;
  final String name;
  final List<Enum> category_type;

  CategoryModel({
    required this.categoryId,
    required this.name,
    required this.category_type,
  });

  factory CategoryModel.fromJson(Map<String, dynamic> json) => CategoryModel(
    categoryId: json['category_id'],
    name: json['name'].toDouble(),
    category_type: json['category_type'],
  );
}